NitroxClient/Communication/Packets/Processors/PlayerSlotReservationProcessor.cs

```diff
@@ -0,0 +1,28 @@
+﻿using NitroxClient.Communication.Packets.Processors.Abstract;
+using NitroxModel.Packets;
+using NitroxModel.PlayerSlot;
+
+namespace NitroxClient.Communication.Packets.Processors
+{
+    public class PlayerSlotReservationProcessor : ClientPacketProcessor<PlayerSlotReservation>
+    {
+        private ClientBridge clientBridge;
+
+        public PlayerSlotReservationProcessor(ClientBridge clientBridge)
+        {
+            this.clientBridge = clientBridge;
+        }
+
+        public override void Process(PlayerSlotReservation packet)
+        {
+            if (packet.ReservationRejectionReason == ReservationRejectionReason.None)
```

Comment:
Enums cannot be null in C#, which means that property would always have a default value that indicated a rejection reason. It is crummy; I know, but is small-beans in the big picture.

An alternative would be to refactor into "ReservationState" with options of "Reservered" or "RejectedOnPlayerName" etc...

This reason code was layered in after I'd already gotten the base functionality implemented, so it really is somewhat half-backed. Again - it was just a good stopping point for the current scope.

Reply:
I'm aware of that fact, and imho both ways aren't optimal. We can leave it for now, as I can't think of a proper alternative.

Or maybe I can.
`ReservationResult`? With the constants listed in your second paragraph.

Reply:
Since it is looking more like we are going to revisit this sooner rather than later - would we want the ability to provide multiple reasons at once? If so, then we can do the rename as you've suggested but also restructure the type to be a bit flag?

Reply:
+1 for bitflags. Then you might even have a the RejectedBecauseOfName be a combination of the Rejected flag and some kind of "NameError" flag.